eodatasets3/assemble.py

```diff
@@ -103,6 +103,7 @@ class DatasetAssembler(EoFields):
         "landsat:collection_number",
         "landsat:landsat_product_id",
         "landsat:landsat_scene_id",
+        "landsat:scene_id",
```

Comment:
We discussed this a bit in #136, but `landsat:landsat_scene_id` and `landsat:scene_id` are the same property, so I'd rather not duplicate them. We already have eo3 products using the first field name.

Previously when the Stac standard changed (with fields like platform, instrument) we kept our existing eo3 fields for consistency across eo3 products, and added the translation in the [Stac converter](https://github.com/GeoscienceAustralia/eo-datasets/blob/5529d5e9a69dea641e86ba7df5ddf2afaf59e8c5/eodatasets3/stac.py#L18-L31).  That way eo3 docs could stay consistent and stable, our existing code would continue to work, and the latest Stac changes would still be present on conversion.

Sticking to the original field name also means we wont need the [new logic below](https://github.com/GeoscienceAustralia/eo-datasets/pull/166/files#diff-50b29b9a54eb3d340c8a094efb3ec9c6f142098d9bffddb47728806d655c2446R439-R447), and we wont need different sets of metadata-types to handle the different field name choices across products ... which historically people have trouble with.

Reply:
Seeing as the USGS STAC docs use `landsat:scene_id`, I think we should include it: https://landsatlook.usgs.gov/sat-api/collections/landsat-c2l2-sr/items

For future reference, properties looks like this:

```
"properties": {
"collection": "landsat-c2l2-sr",
"eo:gsd": 30,
"platform": "LANDSAT_8",
"view:off_nadir": 0,
"datetime": "2021-05-26T22:14:42.761182Z",
"eo:cloud_cover": 83.38,
"view:sun_azimuth": 33.74414161,
"view:sun_elevation": 14.29418577,
"eo:instrument": [
"OLI",
"TIRS"
],
"landsat:cloud_cover_land": 97.39,
"landsat:wrs_type": "2",
"landsat:wrs_path": "074",
"landsat:wrs_row": "093",
"landsat:scene_id": "LC80740932021146LGN00",
"landsat:collection_category": "T1",
"landsat:collection_number": "02"
},
```

Reply:
We might be talking past each-other a bit :slightly_smiling_face: 

My understanding:

- A document can be "eo3" or "stac" but not both. We explicitly have to convert between them already.
- The stac property name has changed from `landsat:landsat_scene_id` to `landsat:scene_id`   (as your point out: USGS uses that field name now), but eo3 documents are already published with the old name.
- This is the same scenario we faced with the `eo:platform`, `eo:instrument` etc fields.

We have two choices:

1. We can start using the new USGS name within `eo3` products, and the `eo3` world will be inconsistent. We will need extra logic to handle the two possible names in every place that reads them, and different ODC metadata types for different products.
or
2. Or we can keep the existing name in the `eo3` world and have our products be consistent. No extra logic, no extra metadata types. When converted to Stac it will convert to the new USGS name.

In _both_ cases our Stac docs will be consistent with USGS. 

In both cases we'll need to add the conversion to our Stac converter (because of old products).

The only decision is whether we want `eo3` to be consistent with itself, or to break it.

I prefer option 2. Our Stac documents will still all be valid and match USGS. We wont need extra code logic to handle both fields everywhere.

(Note: the list of properties in the code we're comment on are `eo3` properties, not Stac properties)

Reply:
We could add this field as a special case here: https://github.com/opendatacube/odc-tools/blob/develop/libs/index/odc/index/stac.py#L17

My issue is that we already have 200,000 datasets indexed in DE Africa and we're about to add 800,000 more.

I guess I could re-index them including changing the `landsat:scene_id` to `landsat:landsat_scene_id` but I really think it's better to be consistent with the USGS metadata, in this case, rather than with Australian EO metadata.

Reply:
Okay, if we've already published datasets with the second name we don't have much choice, we're already using both...

(It wasn't about Australia EO consistency but about ODC system consistency.)